src/coreclr/ilasm/sha256.h

```diff
@@ -0,0 +1,145 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//*****************************************************************************
+// sha256.h
+//
+
+//
+// contains implementation of sha256 hash algorithm
+//
+//*****************************************************************************
+#ifndef __sha256__h__
+#define __sha256__h__
+
+#ifdef _WIN32
+inline HRESULT Sha256Hash(BYTE* pSrc, DWORD srcSize, BYTE* pDst, DWORD dstSize)
+{
+    NTSTATUS status;
+
+    BCRYPT_ALG_HANDLE   algHandle = NULL;
+    BCRYPT_HASH_HANDLE  hashHandle = NULL;
+
+    BYTE    hash[32]; // 256 bits
+    DWORD   hashLength = 0;
+    DWORD   resultLength = 0;
+    status = BCryptOpenAlgorithmProvider(&algHandle, BCRYPT_SHA256_ALGORITHM, NULL, BCRYPT_HASH_REUSABLE_FLAG);
```

Comment:
```suggestion
    status = BCryptOpenAlgorithmProvider(&algHandle, BCRYPT_SHA256_ALGORITHM, NULL, 0);
```
You are not reusing the object. Why pass `BCRYPT_HASH_REUSABLE_FLAG`?

Reply:
The Win32 implementation is a relic from #85344 -- not sure what the original motivation was.